src/app/services/raiden.service.ts

```diff
@@ -61,6 +62,22 @@ export class RaidenService {
         return this._raidenAddress;
     }
 
+    balance$(): Observable<string> {
+        const fetch: () => Observable<string> = () => {
+            return this.raidenAddress$.pipe(
+                flatMap(address =>
+                    fromPromise(this.raidenConfig.web3.eth.getBalance(address))
+                )
+            );
+        };
+        return interval(15000).pipe(
+            startWith(fetch),
+            flatMap(fetch),
+            map(value => fromWei(value, 'ether')),
+            share()
```

Comment:
`share` isn't very useful if you're not dealing with a reference for the same object. You may want to cache the whole observable on a private property to share and return on your getter or whatever, or it'll create a new chain of observables and subscriptions on each access despite `share`

Reply:
Thanks :D